src/vision/calib/Calibration.cpp

```diff
@@ -0,0 +1,210 @@
+                         TermCriteria(TermCriteria::EPS+TermCriteria::COUNT,
+                                      30, 0.1));
+        }
+
+        imagePoints.push_back(pointBuf);
+
+        ROS_DEBUG_STREAM("Drawing corners...");
+        drawChessboardCorners(view, s.boardSize, Mat(pointBuf), found);
+    }
+    else
+    {
+        ROS_DEBUG_STREAM("Pattern not found");
+    }
+
+    string outMsg = (mode == CALIBRATED) ? "Calibrated" : "Calibrating";
+
+    int baseLine = 0;
+    Size textSize = getTextSize(outMsg, 1, 1.0, 1, &baseLine);
+    Point textOrigin(view.cols - 2*textSize.width - 10, 10);
+
+    if (mode == CAPTURING)
+    {
+        ROS_DEBUG_STREAM("Capturing");
+        if (s.showUndistorted)
+        {
+            outMsg = format("%d/%d Undist",
+                            static_cast<int>(imagePoints.size()), s.nrFrames);
+        }
+        else
+        {
+            outMsg = format("%d/%d", static_cast<int>(imagePoints.size()),
+                            s.nrFrames);
+        }
+    }
+
+    if (mode == CALIBRATED && s.showUndistorted)
+    {
+        ROS_DEBUG_STREAM("Undistorting image...");
+        Mat temp = view.clone();
+        if (s.useFisheye)
+        {
+            fisheye::undistortImage(temp, view, cameraMatrix, distCoeffs,
+                                    Matx33d::eye());
+        }
+        else
+        {
+            undistort(temp, view, cameraMatrix, distCoeffs);
+        }
+    }
+
+    ROS_DEBUG_STREAM("Adding text to image");
+    putText(view, outMsg, textOrigin, 1, 1, mode == CALIBRATED ? GREEN : RED);
+
+    imshow("Image", view);
+    char key = static_cast<char>(waitKey(s.delay));
+
+    if (key == ESC_KEY)
+    {
+        ROS_DEBUG_STREAM("User requested stop");
+        calibFinished = true;
+        ros::shutdown();
+        return;
+    }
+
+    if (key == 'u' && mode == CALIBRATED)
+    {
+        s.showUndistorted = !s.showUndistorted;
+        ROS_INFO_STREAM("Now showing " <<
+                        (s.showUndistorted ? "Undistorted" : "Distorted") <<
+                        " image.");
+    }
+}
+
+int main (int argc, char* argv[])
+{
+    ros::init(argc, argv, "camCalib");
+
+    ros::NodeHandle n;
+
+    //Read file
+    if (argc <= 1)
+    {
+        ROS_FATAL("No input settings specified");
+        return 1;
+    }
+
+    ROS_INFO_STREAM("Reading settings file");
+
+    const string inputSettingsFile = argv[1];
+    FileStorage fs(inputSettingsFile, FileStorage::READ);
+    if (!fs.isOpened())
+    {
+        ROS_ERROR_STREAM("Could not open configuration file: " <<
+                         inputSettingsFile);
+        return -1;
+    }
+    ROS_DEBUG_STREAM("File opened");
+
+    fs["Settings"] >> s;
+    ROS_DEBUG_STREAM("Data stored");
+    fs.release();
+
+    if (!s.goodInput)
+    {
+        ROS_ERROR_STREAM("Invalid input detected!");
+        return -1;
+    }
+    ROS_DEBUG_STREAM("Input validated");
```

Comment:
These are insanely ambiguous messages. Change them to specify what input is valid or invalid.